spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1572,7 +1572,10 @@ trait OptimisticTransactionImpl extends TransactionalWrite
 
   private def lockCommitIfEnabled[T](body: => T): T = {
     if (isCommitLockEnabled) {
-      deltaLog.lockInterruptibly(body)
+      // We are borrowing the `snapshotLock` even for commits because multiple threads fighting over
+      // a commit shouldn't interfere with normal snapshot updates by readers. Ideally we should be
+      // using a separate lock for this purpose.
```

Comment:
The comment is worded very confusing... I wonder if part of a sentence ended up in the wrong place?
```suggestion
      // We are borrowing the `snapshotLock` even for commits. Ideally we should be
      // using a separate lock for this purpose, because multiple threads fighting over
      // a commit shouldn't interfere with normal snapshot updates by readers.
```